### DISCUSSION TITLE
Center Window with "Visual Weight"

### DISCUSSION THREAD

**Discussion by vincentpun** [rxhanson/Rectangle#1285] [Ideas]
I recently switched from the Moom app which has a feature called “Center — macOS Center.” Essentially, it does what AppKit’s `NSWindow.center()` does, which, according to [the documentation](https://developer.apple.com/documentation/appkit/nswindow/1419090-center),

> The window is placed exactly in the center horizontally and somewhat above center vertically. Such a placement carries a certain visual immediacy and importance.

It would be nice to have the same feature in Rectangle as well! I have empirically found the formula for calculating the coordinates:

```swift
let windowRect = window.frame
let screenRect = NSScreen.main!.visibleFrame

let centerX = screenRect.center.x
let centerY = screenRect.center.y + (windowRect.height * -0.25 + screenRect.height * 0.25)
```

I can open a PR to add a command called Center with Prominence if this sounds useful!

**Comment by rxhanson** [comment]
Thanks for looking to contribute. I don't mind merging in a PR for this, but at least at first cut it would have to be Terminal command configurable only, with no addition to the UI. I purposefully keep the UI of Rectangle as slim as possible. There are other commands that do this, and you can check out the "extra centering command with custom size" for an example. https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#add-an-extra-centering-command-with-custom-size

**Reply by vincentpun** [reply]
Got it! I created a PR for this: https://github.com/rxhanson/Rectangle/pull/1288
